src/coreclr/src/tools/crossgen2/ILCompiler.ReadyToRun/CodeGen/ReadyToRunObjectWriter.cs

```diff
@@ -99,6 +102,15 @@ public void EmitPortableExecutable()
 
                     ObjectData nodeContents = node.GetData(_nodeFactory);
 
+                    if (node is NativeDebugDirectoryEntryNode nddeNode)
+                    {
+                        // There should is only one NativeDebugDirectoryEntry.
+                        // This assert will need to be revisited when we implement the composite R2R format, where we'll need to figure
+                        // out how native symbols will be emitted, and verify that the DiaSymReader library is able to consume them.
+                        Debug.Assert(nativeDebugDirectoryEntryNode == null);
```

Comment:
cc @trylek FYI

Reply:
Thanks for pointing that out, I'll keep that in mind!